### ISSUE TITLE
ui mask filter

### ISSUE THREAD

**Issue by alexgoroshko** [angular-ui/ui-mask#31]
It would be great to have ui-mask-like filter.
Example:

{{1234567899 | mask: '(999) 999-9999'}} // -> (123) 456-7899

**Comment by chadwhitaker** [comment]
:+1:

**Comment by ssougnez** [comment]
Yeah, totally true. This module sucks without it... Seems quite logical...

**Comment by banzai-samurai** [comment]
+1

**Comment by ssougnez** [comment]
Just found this: http://stackoverflow.com/questions/20999141/how-do-i-get-masked-date-in-model-from-input-in-angular-using-ui-mask

And it works, so it seems it is included... If only stuff such as notifications existed :-D

**Comment by banzai-samurai** [comment]
Thanks, that helped!

**Comment by PowerKiKi** [comment]
PR are welcomed, feel free to improve the demo page.

**Comment by ikr0m** [comment]
+1

**Comment by Tim91084** [comment]
This is interesting.  The use case is slightly different when you're using it as a filter, right?  We wouldn't need to validate, for instance.. And, if we don't need to validate, do we need to define the mask explicitly as containing numbers or letters?  For example, what would be the difference in functionality for the following two filters?

AAA-999-***
***-***-***

For a value of 12A-AB2-A2B...  If A or 9 actually affect the output, what would the output look like?

Assuming they don't affect the output, the only applicable maskDefinition is the *.  If that's the case, this may be a pretty simple filter to write.

**Comment by vinyoliver** [comment]
+!

**Comment by vinyoliver** [comment]
+1

**Comment by hahn-kev** [comment]
I agree, unfortunately it looks like it would require a huge refactor.

**Comment by hahn-kev** [comment]
Well it's not perfect but it should get you by https://gist.github.com/hahn-kev/b3e74a07b53c26ddee1b

**Comment by renatoguarilhadeploy** [comment]
@hahn-kev Thank you very much for that code. You really saved my time. 

Regards

**Comment by H3ll0W0rld** [comment]
+1 much needed feature
